packages/app/src/runner/ResizablePanels.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+  <div
+    class="flex h-full w-full"
+    :class="{'select-none': panel1IsDragging || panel2IsDragging}"
+    @mouseup="handleMouseup"
+    @mousemove="handleMousemove"
+  >
+    <div
+      v-show="showPanel1"
+      class="h-full bg-gray-1000 flex-shrink-0 relative"
+      :style="{width: `${panel1Width}px`}"
+    >
+      <slot
+        name="panel1"
+        :isDragging="panel1IsDragging"
+      />
+
+      <div
+        data-cy="panel1ResizeHandle"
+        class="cursor-ew-resize h-full top-0 -right-6px w-16px z-30 absolute"
+        @mousedown="handleMousedown('panel1', $event)"
+      />
+    </div>
+
+    <div
+      v-show="showPanel2"
+      class="h-full flex-shrink-0 relative"
+      :style="{width: `${panel2Width}px`}"
+    >
+      <slot name="panel2" />
+
+      <div
+        data-cy="panel2ResizeHandle"
+        class="cursor-ew-resize h-full top-0 -right-6px w-16px z-30 absolute"
+        @mousedown="handleMousedown('panel2', $event)"
+      />
+    </div>
+
+    <div
+      class="flex-grow h-full"
+      :class="{'pointer-events-none':panel2IsDragging}"
+      :style="{
+        width: `${maxTotalWidth - panel1Width - panel2Width}px`
+      }"
+    >
+      <slot name="panel3" />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+export default {
+  name: 'ResizablePanels',
+}
+</script>
+
+<script lang="ts" setup>
+import { computed, ref, watchEffect } from 'vue'
+
+const props = withDefaults(defineProps<{
+  showPanel1?: boolean // specsList in runner
+  showPanel2?: boolean // reporter in runner
+  initialPanel1Width?: number
+  initialPanel2Width?: number
+  minPanel1Width?: number
+  minPanel2Width?: number
+  minPanel3Width?: number
+  maxTotalWidth?: number // windowWidth in runner
+  offsetLeft?: number,
+}>(), {
+  showPanel1: true,
+  showPanel2: true,
+  initialPanel1Width: 280,
+  initialPanel2Width: 320,
+  minPanel1Width: 200,
+  minPanel2Width: 220,
+  minPanel3Width: 100,
+  maxTotalWidth: window.innerWidth,
+  offsetLeft: 0,
+})
+
+const emit = defineEmits<{
+  (e: 'resizeEnd', value: 'panel1' | 'panel2'): void,
```

Comment:
I think it's worth having a type for `'panel1' | 'panel2'` since it's used across multiple files. Something like

```
export type AllPanels = 'panel1' | 'panel2' | 'panel3'
export type DraggablePanels = Exclude<AllPanels, 'panel3'>
```